### ISSUE TITLE
BATCH mode first scenario in each node results in different output

### ISSUE THREAD

**Issue by RubenJacobse** [TUDelft-CNS-ATM/bluesky#204]
When running a single scenario multiple times as part of a batch the first run for each node is different from all later runs. 

MD5 checksum of non-batch run (twice repeated using "SAME IC"):
```
FLSTLOG_log_test_20190704_16-06-35.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_log_test_20190704_16-07-22.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_log_test_20190704_16-08-16.log: 144076bb7ce6fd15d8fe791fe974007f
```

MD5 checksums of log files in batch with 1 node:
```
FLSTLOG_test1_20190704_15-45-07.log: 6b8f6e901c223936fb4f2b2070d24f80
FLSTLOG_test2_20190704_15-45-53.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test3_20190704_15-46-38.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test4_20190704_15-47-24.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test5_20190704_15-48-11.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test6_20190704_15-48-59.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test7_20190704_15-49-45.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test8_20190704_15-50-32.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test9_20190704_15-51-18.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test10_20190704_15-52-05.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test11_20190704_15-52-53.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test12_20190704_15-53-39.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test13_20190704_15-54-26.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test14_20190704_15-55-12.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test15_20190704_15-55-58.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test16_20190704_15-56-43.log: 144076bb7ce6fd15d8fe791fe974007f
```

MD5 checksums of log files in batch with 4 nodes:
```
FLSTLOG_test1_20190704_15-39-24.log: 6b8f6e901c223936fb4f2b2070d24f80
FLSTLOG_test2_20190704_15-39-25.log: 6b8f6e901c223936fb4f2b2070d24f80
FLSTLOG_test3_20190704_15-39-25.log: 6b8f6e901c223936fb4f2b2070d24f80
FLSTLOG_test4_20190704_15-39-25.log: 6b8f6e901c223936fb4f2b2070d24f80
FLSTLOG_test5_20190704_15-40-22.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test6_20190704_15-40-22.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test7_20190704_15-40-22.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test8_20190704_15-40-22.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test9_20190704_15-41-20.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test10_20190704_15-41-21.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test11_20190704_15-41-21.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test12_20190704_15-41-21.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test13_20190704_15-42-18.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test14_20190704_15-42-18.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test15_20190704_15-42-18.log: 144076bb7ce6fd15d8fe791fe974007f
FLSTLOG_test16_20190704_15-42-18.log: 144076bb7ce6fd15d8fe791fe974007f
```

Scenario files used:
[batch_log_test.txt](https://github.com/TUDelft-CNS-ATM/bluesky/files/3359642/batch_log_test.txt)
[log_test.txt](https://github.com/TUDelft-CNS-ATM/bluesky/files/3359643/log_test.txt)

**Comment by jooste** [comment]
Hi Ruben,

I don't get any differences. Which performance model do you use?

**Comment by RubenJacobse** [comment]
Hi Joost,

I used the legacy performance model

**Comment by jooste** [comment]
try switching to the OpenAP to see if you get the same results. Hopefully that narrows it down!

**Comment by RubenJacobse** [comment]
Using OpenAP results in different hashes (as expected), but the same pattern persists.

MD5 checksums of log files in batch with 4 nodes:
```
FLSTLOG_test1_20190704_17-19-12.log: 8fb3d21c48cff5250f16a38bf75e0d4d
FLSTLOG_test2_20190704_17-19-15.log: 8fb3d21c48cff5250f16a38bf75e0d4d
FLSTLOG_test3_20190704_17-19-15.log: 8fb3d21c48cff5250f16a38bf75e0d4d
FLSTLOG_test4_20190704_17-19-15.log: 8fb3d21c48cff5250f16a38bf75e0d4d
FLSTLOG_test5_20190704_17-19-56.log: f22cd3ef5f58983de877e2eac8000596
FLSTLOG_test6_20190704_17-19-58.log: f22cd3ef5f58983de877e2eac8000596
FLSTLOG_test7_20190704_17-19-58.log: f22cd3ef5f58983de877e2eac8000596
FLSTLOG_test8_20190704_17-19-59.log: f22cd3ef5f58983de877e2eac8000596
FLSTLOG_test9_20190704_17-20-43.log: f22cd3ef5f58983de877e2eac8000596
FLSTLOG_test10_20190704_17-20-45.log: f22cd3ef5f58983de877e2eac8000596
FLSTLOG_test11_20190704_17-20-45.log: f22cd3ef5f58983de877e2eac8000596
FLSTLOG_test12_20190704_17-20-46.log: f22cd3ef5f58983de877e2eac8000596
FLSTLOG_test13_20190704_17-21-30.log: f22cd3ef5f58983de877e2eac8000596
FLSTLOG_test14_20190704_17-21-31.log: f22cd3ef5f58983de877e2eac8000596
FLSTLOG_test15_20190704_17-21-31.log: f22cd3ef5f58983de877e2eac8000596
FLSTLOG_test16_20190704_17-21-32.log: f22cd3ef5f58983de877e2eac8000596
```

**Comment by RubenJacobse** [comment]
Did some more testing by adding print statements. There seems to be a time offset issue for plugins with initialization of scenarios called via a batch file.

When a scenario file is opened directly all stack commands set for execution at `"0:00:00.00"` are indeed executed at this time (`bs.sim.simt = 0.0`). However, when a scenario is called via a batch file only the stack commands of the calling batch file are executed (the contents of the batch event) and the execution of the actual scenario stack commands is delayed until the next time step. 

This causes problems when using plugins such as `area.py` (or my own plugin) because these are now also initialized and triggered at an offset of a single time step for the first scenario. For the second scenario in the batch the `sim.reset()` call causes the already initialized plugin to be reset as well which resets the plugin triggering offset.

**Comment by jooste** [comment]
Hi, this should be fixed in 7d49e94.
